python/modem-firmware-1.3+/nrf_cloud_provision.py

```diff
@@ -66,33 +76,56 @@ def parse_args():
     args = parser.parse_args()
     return args
 
+def set_dev_stage(stage = ''):
+    global api_url
+    global dev_stage_key
+
+    if stage in DEV_STAGE_DICT.keys():
+        dev_stage_key = stage
+        api_url = '{}{}{}'.format(API_URL_START, DEV_STAGE_DICT[dev_stage_key], API_URL_END)
+    else:
+        print('Invalid stage')
+
+    return api_url
+
 def get_bulk_ops_result(api_key, bulk_ops_req_id):
     hdr = {'Authorization': 'Bearer ' + api_key}
-    req = API_URL + "bulk-ops-requests/" + bulk_ops_req_id
+    req = api_url + "bulk-ops-requests/" + bulk_ops_req_id
     return requests.get(req, headers=hdr)
 
 def update_device_shadow(api_key, device_id, json_payload):
     hdr = {'Authorization': 'Bearer ' + api_key}
-    req = API_URL + "devices/" + device_id + "/state"
+    req = api_url + "devices/" + device_id + "/state"
     return requests.patch(req, json=json_payload, headers=hdr)
 
 def fetch_device(api_key, device_id):
     hdr = {'Authorization': 'Bearer ' + api_key}
-    req = API_URL + "devices/" + device_id
+    req = api_url + "devices/" + device_id
     return requests.get(req, headers=hdr)
 
 def update_device_name(api_key, device_id, name):
     hdr = {'Authorization': 'Bearer ' + api_key}
-    req = API_URL + "devices/" + device_id + "/name"
+    req = api_url + "devices/" + device_id + "/name"
     json_payload = [name]
     return requests.put(req, json=json_payload, headers=hdr)
 
+def provision_device(api_key, dev_id, sub_type, tags, fw_types, cert_pem_str):
+    hdr = {'Authorization': 'Bearer ' + api_key,
```

Comment:
Seems like this could allow a non-Nordic person to provision a device to dev or beta.  Maybe not a new problem, but perhaps a security risk?

Reply:
if they have an API key for dev or beta, they can do whatever they want :)